### PR TITLE
Add optional datastax request logger

### PR DESCRIPTION
**Pull request by porunov**
Adds possibility to enable CQL request logger to debug queries. Tested manually.
Not sure how to cover this feature with tests without using reflection (which might be a bad approach).

Request tracker and a builtin request logger DataStax documentation for 4.9 driver version is here: https://docs.datastax.com/en/developer/java-driver/4.9/manual/core/request_tracker/

Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLConfigOptions.java [hunk 1]
```diff
@@ -285,4 +285,44 @@
...
+        "Registers a query logger with datastax cassandra driver",
+        ConfigOption.Type.LOCAL,
+        false);
+
+    ConfigOption<Integer> QUERY_LOGGER_MAX_QUERY_STRING_LENGTH = new ConfigOption<>(
+        QUERY_LOGGER,
+        "max-query-string-length",
+        "Max query string length to be logged when query logger is enabled. Use '-1' to use datastax default value.",
+        ConfigOption.Type.LOCAL,
+        -1);
```
Comment (farodin91) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLConfigOptions.java:306]: It would be okay to have on default value.

Comment (farodin91) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLConfigOptions.java:306]: https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java#L394

**Comment by farodin91** [comment]
which might be a bad approach knowing that we are going to upgrade our CQL driver soon. This upgrade isn't that easy.

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java [hunk 2]
```diff
@@ -316,7 +281,35 @@ Cluster initializeCluster() throws PermanentBackendException {
                     HostDistance.REMOTE,
                     configuration.get(REMOTE_CORE_CONNECTIONS_PER_HOST),
                     configuration.get(REMOTE_MAX_CONNECTIONS_PER_HOST));
-        return builder.withPoolingOptions(poolingOptions).build();
+
+        Cluster cluster = builder.withPoolingOptions(poolingOptions).build();
+
+        if(configuration.get(QUERY_LOGGER_ENABLED)){
```
Comment (farodin91) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java:287]: You forgot a space in each if.

**Comment by porunov** [comment]
@farodin91 Queries logging is configured a little bit differently in version 4.x of datastax cassandra driver (https://docs.datastax.com/en/developer/java-driver/4.7/manual/core/request_tracker/#request-logger), so I think we need to hold off of this feature. I will rebase it when we done upgrading datastax cassandra driver

**Comment by farodin91** [comment]
@porunov do you want to work on the new cql driver?

**Comment by porunov** [comment]
@farodin91 
I've made a draft PR #2169 . Didn't test and didn't go deep on changes in that PR. Will work on it later but early review is very welcome.

**Comment by porunov** [comment]
@farodin91 , I rebased against 4.x driver (`master` branch). Tested manually, all configurations work as expected. Also, tested this PR with custom integration tests - all tests passed. Resolved your current comments. A new review would be appreciated.

**Review by farodin91 — APPROVED** [review]
@porunov LGTM! Thank you for your work.

nit: Don' t think we need any complex test. I would add a test that ensures basic are working after activating the request tracker.

**Comment by porunov** [comment]
Thank you @farodin91 ! I've amended the commit with the test here: https://github.com/JanusGraph/janusgraph/compare/5d375eff357e63a855c1210e2dd9867faf465d55..3e6e09098e72ef54b55495f1b758b80bc803744f

**Pull request merged**